src/components/toast-notifications/toastNotifications.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+angular.module('openshiftCommonUI')
+  .directive('toastNotifications', function() {
+    return {
+      restrict: 'E',
+      scope: {
+        // Map (or array) of alert objects, each with properties
+        //   type:     'error', 'warning', 'success', or 'info' (default)
+        //   message:  the message to display
+        //   links:    an array of link objects (optional, see below)
+        //   onClose:  callback function when the user dismisses the alert (optional)
+        //
+        // Each link can have the following properties:
+        //   href:     link URL (optional if onClick is set)
+        //   label:    the link text
+        //   onClick:  handler to call when the link if clicked (optional)
+        //             If onClick returns true, dismisses the alert
+        alerts: '=',
+        filter: '=?',
+        hideCloseButton: '=?'
+      },
+      templateUrl: 'src/components/toast-notifications/toast-notifications.html',
+      link: function($scope) {
+        $scope.close = function(alert) {
+          alert.hidden = true;
+          if (_.isFunction(alert.onClose)) {
+            alert.onClose();
+          }
+        };
+        $scope.onClick = function(alert, link) {
+          if (_.isFunction(link.onClick)) {
+            // If onClick() returns true, also hide the alert.
+            var close = link.onClick();
+            if (close) {
+              alert.hidden = true;
+            }
+          }
+        };
+        $scope.setHover = function(alert, isHover) {
+          alert.isHover = isHover;
```

Comment:
This will be used by applications. Some toast notification may auto dismiss (typically the success notifications) however they should not dismiss if the user is currently hovered over the toast as that is an indication that they are currently looking at the notification.